python/cudf/dataframe/dataframe.py

```diff
@@ -1162,9 +1161,25 @@ def fix_name(name, suffix):
                                               mask=Buffer(valids[idx]))
                 idx = idx + 1
 
+        # Creating dataframe with ordering as pandas:
+        # Pandas always orders the output first according to the input
+        # column order.
+
+        final_df = DataFrame()
+        for col in lhs.columns:
+            fcol = col
+            if col not in df.columns:
+                fcol = fix_name(col, lsuffix)
+            if fcol not in df.columns:
+                fcol = fix_name(col, rsuffix)
+            final_df[fcol] = df[fcol]
+        for col in df.columns:
+            if col not in final_df.columns:
+                final_df[col] = df[col]
```

Comment:
Rather than grabbing the columns from `df` this logic should be moved above into where `df` creates its Series objects.